portfolio/src/main/webapp/script.js

```diff
@@ -50,9 +50,26 @@ function showPicture(){
   imageContainer.appendChild(imgElement);    
 }
 
+/**
+ * Alerts the user when they have failed to fill out all required
+ * input fields.
+ */
+function required() {
+    var nameElement = document.forms["form1"]["user-name"].value;
+    var commentElement = document.forms["form1"]["user-comment"].length;
+    console.log(commentElement);
+    if (nameElement == "" || nameElement == null){
```

Comment:
Missing the check of commentElement.
You can implement a util function e.g:
function isEmpty(str) {
    return (!str || str === "");
}

then use it:
 if (isEmpty(nameElement) || isEmpty(commentElement))

It will be more clear.